be/src/storage/lake/persistent_index_memtable.cpp

```diff
@@ -16,82 +16,110 @@
 
 namespace starrocks::lake {
 
+void PersistentIndexMemtable::insert(const std::string& key, int64_t version, const IndexValue& value) {
+    std::list<IndexValueInfo> index_value_infos;
+    index_value_infos.emplace_front(version, value);
+    _map.emplace(key, index_value_infos);
+}
+
+void PersistentIndexMemtable::update(std::list<IndexValueInfo>& index_value_infos, int64_t version,
+                                     const IndexValue& value) {
+    std::list<IndexValueInfo> t;
+    t.emplace_front(version, value);
+    index_value_infos.swap(t);
+}
+
 Status PersistentIndexMemtable::upsert(size_t n, const Slice* keys, const IndexValue* values, IndexValue* old_values,
-                                       KeyIndexesInfo* not_found, size_t* num_found) {
+                                       KeyIndexesInfo* not_found, size_t* num_found, int64_t version) {
```

Comment:
How about defining `keys` as type  `std::span<const Slice>`?

Reply:
It seems that const Slice* can not be casted to std::span<const Slice>